### PR TITLE
Batched messages and ecrecover

### PR DESCRIPTION
**Pull request by edfelten**
* Implement ecrecover AVM instruction
* Process aggregated batches of txs in ArbOS
* Generate signed txs on the client side, and verify them (in txs from batches) in ArbOS
* Fix bug in ArbSys precompile
* Add ecrecover precompiled contract at address 1
* Refactor handling of precompiles
* Remove erroneous tests

### CODE REVIEW

File: arb_os/precompiles.mini [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    acctStore = accountStore_createBuiltinContract(
+        acctStore,
+        address(100),
+        arbsys_txcall,
+    )?;
+    return Some(acctStore);
+}
+
+public impure func precompile_0x01() {  // ecrecover
+    // don't bother checking the first four bytes of calldata -- this is the only method in the contract
```
Comment (hkalodner) [inline arb_os/precompiles.mini:57]: From the yellow paper:

h = d[0..31]
v = d[32..63]
r = d[64..95]
s = d[96..127]

File: src/evm/mod.rs [hunk 2]
```diff
@@ -178,11 +179,14 @@ pub fn evm_xcontract_call_using_batch(
 ) -> Result<bool, ethabi::Error> {
     use std::convert::TryFrom;
     let rt_env = RuntimeEnvironment::new(Uint256::from_usize(1111));
+
+    let wallet = rt_env.new_wallet();
+    let addr_h160: Address = wallet.private_key().into();
+    let my_addr = Uint256::from_bytes(addr_h160.as_bytes());
```
Comment (hkalodner) [inline src/evm/mod.rs:185]: Is wallet of type `ethers_signers::Wallet`? If so it seems like we should just use `wallet.address()` here

Comment (edfelten) [reply src/evm/mod.rs:185]: Switched to using `wallet.address()`.  But I still need to convert it to a Uint256.

File: arb_os/precompiles.mini (hunk 1 above)
Comment (edfelten) [reply arb_os/precompiles.mini:57]: Changed to match the yellow paper.

Comment (hkalodner) [reply arb_os/precompiles.mini:57]: Sorry, to be clear I think that based on the spec it looks like it starts from an offset of 0 rather than 4.

Comment (edfelten) [reply arb_os/precompiles.mini]: Fixed. Thanks for pointing this out.

**Comment by edfelten** [comment]
Last change was trivial, and Harry apparently approved merge, so I'm merging this.

**Comment by hkalodner** [comment]
+1 😁

**Pull request merged**